### PR TITLE
Add simple authorization

### PR DESCRIPTION
**Pull request by Tzumx**

### CODE REVIEW

File: routers/users.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    user = await users_crud.get_user_by_email(email=form_data.username)
+
+    if not user:
+        raise HTTPException(status_code=400, detail="Incorrect email or password")
+
+    if not users_util.validate_password(
+        password=form_data.password, hashed_password=user["hashed_password"]
+    ):
+        raise HTTPException(status_code=400, detail="Incorrect email or password")
```
Comment (mikaserova) [inline routers/users.py]: 401 error code

File: routers/users.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                parameters required to login a user
+        Returns:
+            response: User
+                instance of logged user's token
+    """
+
+    user = await users_crud.get_user_by_email(email=form_data.username)
+
+    if not user:
+        raise HTTPException(status_code=400, detail="Incorrect email or password")
```
Comment (mikaserova) [inline routers/users.py]: 401 error code

File: routers/users.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    if not users_util.validate_password(
+        password=form_data.password, hashed_password=user["hashed_password"]
+    ):
+        raise HTTPException(status_code=400, detail="Incorrect email or password")
+
+    return await users_crud.create_user_token(user_id=user["id"])
```
Comment (mikaserova) [inline routers/users.py]: you do not need to store user password in DB, you should generate it on auth and return to user. And then with any endpoint call decode jwt token, and validate it https://www.freecodecamp.org/news/how-to-add-jwt-authentication-in-fastapi/

**Review by mikaserova — COMMENTED** [review]
Do not store tokens in DB, use classic JWT Auth  flow

File: routers/users.py (hunk 1 above)
Comment (Tzumx) [reply routers/users.py]: Done

File: routers/users.py (hunk 2 above)
Comment (Tzumx) [reply routers/users.py]: Done

File: routers/users.py (hunk 3 above)
Comment (Tzumx) [reply routers/users.py]: Done

**Review by mikaserova — APPROVED** [review]
Please add authentication to all protected endpoints in follow up PR

**Pull request merged**